.github/workflows/ci.yml

```diff
@@ -0,0 +1,186 @@
+            run: |
+              sudo apt-get install shellcheck
+              ./run-tests.sh --check-shellscript
+
+    lint-black:
+        runs-on: ubuntu-latest
+        steps:
+          - uses: actions/checkout@v2
+
+          - name: Setup Python
+            uses: actions/setup-python@v2
+            with:
+              python-version: 3.8
+
+          - name: Check Python code formatting
+            run: |
+              pip install black
+              ./run-tests.sh --check-black
+
+    lint-pydocstyle:
+      runs-on: ubuntu-latest
+      steps:
+        - uses: actions/checkout@v2
+
+        - name: Setup Python
+          uses: actions/setup-python@v2
+          with:
+            python-version: 3.8
+
+        - name: Check compliance with Python docstring conventions
+          run: |
+            pip install pydocstyle
+            ./run-tests.sh --check-pydocstyle
+
+    lint-check-manifest:
+      runs-on: ubuntu-latest
+      steps:
+        - uses: actions/checkout@v2
+
+        - name: Setup Python
+          uses: actions/setup-python@v2
+          with:
+            python-version: 3.8
+
+        - name: Check Python manifest completeness
+          run: |
+            pip install check-manifest
+            ./run-tests.sh --check-manifest
+
+    docs-cli-commands:
+        runs-on: ubuntu-latest
+        steps:
+          - uses: actions/checkout@v2
+    
+          - name: Setup Python
+            uses: actions/setup-python@v2
+            with:
+                python-version: 3.8
+    
+          - name: Install Python dependencies
+            run: |
+                pip install --upgrade pip setuptools py
+                pip install -e .[all]
+    
+          - name: Verify reana client commands list
+            run: ./run-tests.sh --check-cli-cmds
+
+    docs-cli-api:
+        runs-on: ubuntu-latest
+        steps:
+          - uses: actions/checkout@v2
+    
+          - name: Setup Python
+            uses: actions/setup-python@v2
+            with:
+                python-version: 3.8
+
+          - name: Install Python dependencies
+            run: |
+                pip install --upgrade pip setuptools py
+                pip install -e .[all]
+
+          - name: Verify reana client api docs
+            run: ./run-tests.sh --check-cli-api
+
+    docs-sphinx:
+      runs-on: ubuntu-latest
+      steps:
+        - uses: actions/checkout@v2
+
+        - name: Setup Python
+          uses: actions/setup-python@v2
+          with:
+            python-version: 3.8
+
+        - name: Install Python dependencies
+          run: |
+            pip install --upgrade pip setuptools py
+            pip install -e .[all]
+
+        - name: Run Sphinx documentation with doctests
+          run: ./run-tests.sh --check-sphinx
+
+    python-tests:
+        runs-on: ubuntu-latest
+        strategy:
+          matrix:
+              include:
+                - testenv: lowest
```

Comment:
Is there any specific reason for using this approach over `python-version: [...]` as in [Audrius PR](https://github.com/reanahub/reana/pull/433/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR104)? I find it more compact and readable.

Reply:
Yes, here we have 2 different environments to test in `lowest` and `release` and these envs have to be in each python version.

I tried `python-version` but then it gives priority to it and not `testenv`.